readthedocs/redirects/models.py

```diff
@@ -324,6 +327,20 @@ def redirect_clean_url_to_html(self, filename, path, language=None, version_slug
                     allow_crossdomain=False,
                 )
 
+    def redirect_clean_url_without_trailing_slash_to_html(
+        self, filename, path, language=None, version_slug=None
+    ):
+        log.debug("Redirecting...", redirect=self)
+        name, ext = os.path.splitext(filename)
+        if not ext:
+            to = name + ".html"
+            return self.get_full_path(
```

Comment:
This still feels weird to me. It _will_ redirect `/foo` to `/foo.html`, even if `/foo.html` doesn't exist. I thought the other implementations here only redirected if this file existed? It seems OK, but very unexpected as a user and maintainer :/

Reply:
That's how redirect works, right? If the user defines any redirect we don't check the `to URL` exists before redirecting. However, we do check if the `from URL` exists before redirecting, unless `force=True` is defined.

https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/redirects/models.py#L310-L335